components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediator.java

```diff
@@ -0,0 +1,411 @@
+                    messageContext.getEnvelope().getBody().addChild(omElement);
+                }
+            } else {
+                messageContext.setEnvelope(createDefaultSOAPEnvelope(messageContext));
+            }
+        } catch (DataServiceFault dataServiceFault) {
+            if (synLog.isTraceOrDebugEnabled()) {
+                synLog.traceOrDebug(dataServiceFault.getMessage());
+            }
+            synLog.error(dataServiceFault.getMessage());
+            handleException("DataService exception occurred while accessing the dataservice to do the operation",
+                    dataServiceFault, messageContext);
+
+        } catch (AxisFault axisFault) {
+            if (synLog.isTraceOrDebugEnabled()) {
+                synLog.traceOrDebug(axisFault.getMessage());
+            }
+            handleException("Error while creating response payload", axisFault, messageContext);
+        }
+    }
+
+    private void addOperations(Operations operations, OMElement payload, MessageContext msgCtx) {
+
+        for (Object operationObj : operations.getOperations()) {
+            if (operationObj instanceof Operation) {
+                Operation nestedOperation = (Operation) operationObj;
+                String operationName = nestedOperation.getOperationName();
+                OMElement nestedPayload = fac.createOMElement(operationName, omNamespace);
+                for (Param param : nestedOperation.getParams()) {
+                    addParams(param, nestedPayload, msgCtx);
+                }
+                payload.addChild(nestedPayload);
+            } else if (operationObj instanceof Operations) {
+                Operations rootOperations = (Operations) operationObj;
+                String rootOpName;
+                if (rootOperations.getType() == OperationsType.SINGLE_REQ) {
+                    Operation singleOp = (Operation) rootOperations.getOperations().get(0);
+                    rootOpName = singleOp.getOperationName();
+                } else {
+                    rootOpName = rootOperations.getType().toString().toLowerCase();
+                }
+                OMElement nestedOpEle = fac.createOMElement(rootOpName, omNamespace);
+                addOperations(rootOperations, nestedOpEle, msgCtx);
+                payload.addChild(nestedOpEle);
+            }
+        }
+    }
+
+    private void addParams(Param param, OMElement payload, MessageContext msgCtx) {
+        OMElement omElement = fac.createOMElement(param.getParamName(), omNamespace);
+        String paramValue = "";
+        if (param.getParamValue() != null) {
+            paramValue = param.getParamValue();
+        } else if (param.getParamExpression() != null) {
+            paramValue = param.getParamExpression().stringValueOf(msgCtx);
+            if (DataServiceCallMediatorConstants.XML_TYPE.equals(param.getParamType()) &&
+                    !isJson(paramValue.trim(), param.getParamExpression())) {
```

Comment:
can we not specify json-eval

Reply:
We can use. 
<param name="name" type="string" evaluator="json"  expression="$.test"/>